core-bundle/src/Search/Backend/Provider/FilesProvider.php

```diff
@@ -27,10 +30,16 @@ class FilesProvider implements ProviderInterface
 {
     public const TYPE = 'file';
 
+    private readonly PermissionCheckingVirtualFilesystem $permissionCheckingFilesStorage;
+
     public function __construct(
-        private readonly Connection $connection,
-        private readonly AccessDecisionManagerInterface $accessDecisionManager,
+        private readonly VirtualFilesystemInterface $filesStorage,
+        Security $security,
+        private readonly Studio $studio,
     ) {
+        $this->permissionCheckingFilesStorage = new PermissionCheckingVirtualFilesystem(
```

Comment:
Don't we have that as a service already? Just in case its parameters once change because it needs more info than just `Security`?

Reply:
See e https://github.com/contao/contao/pull/7637